python/cuspatial/cuspatial/tests/binpreds/test_equals_only_binops.py

```diff
@@ -0,0 +1,570 @@
+    line = cuspatial.from_geopandas(gpdline)
+    polygon = cuspatial.from_geopandas(gpdpolygon)
+    got = polygon.geom_equals(line)
+    expected = gpdpolygon.geom_equals(gpdline)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_polygons_geom_equals_10000_linestrings(
+    polygon_generator, linestring_generator
+):
+    gpdpolygons = gpd.GeoSeries([*polygon_generator(10000, 0)])
+    gpdlines = gpd.GeoSeries([*linestring_generator(10000, 5)])
+    polygons = cuspatial.from_geopandas(gpdpolygons)
+    lines = cuspatial.from_geopandas(gpdlines)
+    got = polygons.geom_equals(lines)
+    expected = gpdpolygons.geom_equals(gpdlines)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_polygon_geom_equals_polygon():
+    gpdpolygon1 = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
+    gpdpolygon2 = gpd.GeoSeries(Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]))
+    polygon1 = cuspatial.from_geopandas(gpdpolygon1)
+    polygon2 = cuspatial.from_geopandas(gpdpolygon2)
+    got = polygon1.geom_equals(polygon2)
+    expected = gpdpolygon1.geom_equals(gpdpolygon2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+@pytest.mark.parametrize(
+    "lhs",
+    [
+        [
+            Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+        ],
+        [
+            Polygon([[0, 0], [2, 0], [2, 2], [0, 0]]),
+            Polygon([[0, 0], [2, 0], [2, 2], [0, 0]]),
+            Polygon([[0, 0], [2, 0], [2, 2], [0, 0]]),
+        ],
+        [
+            Polygon([[0, 0], [3, 0], [3, 3], [0, 0]]),
+            Polygon([[0, 0], [3, 0], [3, 3], [0, 0]]),
+            Polygon([[0, 0], [3, 0], [3, 3], [0, 0]]),
+        ],
+    ],
+)
+def test_3_polygons_geom_equals_3_polygons_one_equal(lhs):
+    gpdpolygons1 = gpd.GeoSeries(lhs)
+    gpdpolygons2 = gpd.GeoSeries(
+        [
+            Polygon([[0, 0], [1, 0], [1, 1], [0, 0]]),
+            Polygon([[0, 0], [2, 0], [2, 2], [0, 0]]),
+            Polygon([[0, 0], [3, 0], [3, 3], [0, 0]]),
+        ]
+    )
+    polygons1 = cuspatial.from_geopandas(gpdpolygons1)
+    polygons2 = cuspatial.from_geopandas(gpdpolygons2)
+    got = polygons1.geom_equals(polygons2)
+    expected = gpdpolygons1.geom_equals(gpdpolygons2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_10000_polygons_geom_equals_10000_polygons(polygon_generator):
+    gpdpolygons1 = gpd.GeoSeries([*polygon_generator(10000, 0)])
+    gpdpolygons2 = gpd.GeoSeries([*polygon_generator(10000, 0)])
+    polygons1 = cuspatial.from_geopandas(gpdpolygons1)
+    polygons2 = cuspatial.from_geopandas(gpdpolygons2)
+    got = polygons1.geom_equals(polygons2)
+    expected = gpdpolygons1.geom_equals(gpdpolygons2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_point_contains_point():
```

Comment:
All test below this points are testing `contains_properly` I suppose?

Reply:
No, these binary predicates depend only on `geom_equals` to return correct results. A point can contain a point in GeoPandas, and it appears to be a simple equality test.